src/Verilator.cpp

```diff
@@ -590,15 +600,25 @@ int main(int argc, char** argv, char** env) {
     V3Global::dumpCheckGlobalTree("final", 990, v3Global.opt.dumpTreeLevel(__FILE__) >= 3);
     V3Error::abortIfWarnings();
 
-    if (v3Global.opt.makeDepend().isTrue()) {
-        V3File::writeDepend(v3Global.opt.makeDir()+"/"+v3Global.opt.prefix()+"__ver.d");
-    }
-    if (v3Global.opt.skipIdentical().isTrue() || v3Global.opt.makeDepend().isTrue()) {
-        V3File::writeTimes(v3Global.opt.makeDir()+"/"+v3Global.opt.prefix()
-                           +"__verFiles.dat", argString);
-    }
-    if (v3Global.opt.protectIds()) {
-        VIdProtect::writeMapFile(v3Global.opt.makeDir()+"/"+v3Global.opt.prefix()+"__idmap.xml");
+    if (v3Global.hierPlanp()) {  // This run is for just write a makefile
+        if (v3Global.opt.cmake()) {
+            //V3EmitCMake::emitHierVerilation(v3Global.hierPlanp());
+            UASSERT(false, "Not implemented yet");
+        }
+        if (v3Global.opt.gmake()) {
+            V3EmitMk::emitHierVerilation(v3Global.hierPlanp());
+        }
+    } else {
+        if (v3Global.opt.makeDepend().isTrue()) {
+            V3File::writeDepend(v3Global.opt.makeDir()+"/"+v3Global.opt.prefix()+"__ver.d");
+        }
+        if (v3Global.opt.skipIdentical().isTrue() || v3Global.opt.makeDepend().isTrue()) {
+            V3File::writeTimes(v3Global.opt.makeDir()+"/"+v3Global.opt.prefix()
+                               +"__verFiles.dat", argString);
+        }
```

Comment:
If using a hier plan and the inputs are identical can we skip some steps?

Reply:
Yes, make will skip Verilation of the hierarchy blocks.